### PR TITLE
Add Minesweeper exercise

### PR DESCRIPTION
**Pull request by devkabiir**
This exercise:
- Focuses on classes as data types
- Logic and mathematics
- Has a hints section in readme.md

I've left the config.json untouched purposely, need opinions before I can add this to config.
What difficulty, unlocked by, etc. should be.

### CODE REVIEW

**Comment by Stargator** [comment]
Just about the difficulty, this is difficult problem, I'm not sure we have the right tests that could lead up to it.

For example, [the Java track rates it as a 6 difficulty](https://github.com/exercism/java/blob/c8ae482641f54290142c32b9c3493ee5f3c8abbc/config.json#L614).

**Comment by Stargator** [comment]
[The specification for this exercise was updated](https://github.com/exercism/problem-specifications/pull/1124) to fit a new definition, double check that there aren't any additional changes

**Comment by devkabiir** [comment]
There aren't any new tests added in exercism/problem-specifications/pull/1124. Only schema has changed,I don't think if this needs any updates. Although the config.json is till untouched for this one.

**Comment by Stargator** [comment]
For the config.json:

Not a core exercise, I would put it at a 5 difficulty, but that's just my guessing. And just add the exercise to bottom of the file. Whatever topics you feel are appropriate.

**Comment by Stargator** [comment]
Hey @devkabiir do you think you can complete this by April 30th? That's the deadline for the New Batch of Exercises milestone

**Comment by devkabiir** [comment]
I'm not updated to the latest changes. Does this need anything else updated?

File: exercises/minesweeper/test/minesweeper_test.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import 'package:test/test.dart';
+import 'package:minesweeper/minesweeper.dart';
```
Comment (Stargator) [inline exercises/minesweeper/test/minesweeper_test.dart]: If you don't mine sorting the imports to be alphabetical.

**Pull request merged**